src/symbolic_dimensions.jl

```diff
@@ -86,12 +137,12 @@ for (type, _, _) in ABSTRACT_QUANTITY_TYPES
             dims = SymbolicDimensions{R}(I, V)
             return constructorof(Q)(convert(T, ustrip(q)), dims)
         end
-        function Base.convert(::Type{Q}, q::UnionAbstractQuantity{<:Any,<:SymbolicDimensions}) where {T,D<:Dimensions,Q<:$type{T,D}}
+        function Base.convert(::Type{Q}, q::UnionAbstractQuantity{<:Any,<:AbstractSymbolicDimensions}) where {T,D<:Dimensions,Q<:$type{T,D}}
             result = constructorof(Q)(convert(T, ustrip(q)), D())
             d = dimension(q)
-            for (idx, value) in zip(getfield(d, :nzdims), getfield(d, :nzvals))
+            for (idx, value) in zip(nzdims(d), nzvals(d))
                 if !iszero(value)
-                    result = result * convert(with_type_parameters(Q, T, D), ALL_VALUES[idx]) ^ value
+                    result *= convert(with_type_parameters(Q, T, D), ALL_VALUES[idx] ^ value)
                 end
```

Comment:
This could be simplified for the `...Singleton` type.

Reply:
Just checking – the compiler can probably simplify this itself now that I made the `nzvals` and `nzdims` tuples, right?